### PR TITLE
Add diagnostic.md and diagnostic.js with responses

### PR DESCRIPTION
**Pull request by DanielJRaine**
Comfort: 5
Clarity: 4

I used Google and my in class work to help with this assessment.

### CODE REVIEW

**Comment by DanielJRaine** [comment]
Comfort: 5
Clarity: 4

I used Google and my in class work to help with this assessment.

File: diagnostic.md [hunk 1]
```diff
@@ -9,6 +9,12 @@ Assume all code executes in strict mode.
 
 Please list the JavaScript primitives and give one example of each.
 
+number = 1
+boolean = true
+string = 'hello world'
+null = null (an empty set.  Falsey.  Not zero or NaN.  Just null)
+undefined = undefined (declaring a variable without assigning it: let x;
```
Comment (berziiii) [inline diagnostic.md:16]: Correct.

File: diagnostic.md [hunk 2]
```diff
@@ -20,6 +26,9 @@ c = c + d;
 
 After this code executes, what is the value of c?  Please identify the operators in the last line of this program `c = c + d;` and explain what they do.
 
+c === 7 by the end of the program.
```
Comment (berziiii) [inline diagnostic.md:29]: Correct.

File: diagnostic.md [hunk 3]
```diff
@@ -32,19 +41,28 @@ y = 10;
...
-<!-- solution below -->
+let x = 4 (x is undefined)
+x is 4
+
+let y = 3 (y is undefined)
+          (y is 3, x is 4)
 
+x = y (x is 3, y is 3)
 
+y = 10 (x is 3, y is 10)
```
Comment (berziiii) [inline diagnostic.md:52]: Correct.

File: diagnostic.md [hunk 4]
```diff
@@ -32,19 +41,28 @@ y = 10;
...
+
+The values of the variables change whenever we assign or reassign them.  The variables themselves do not change, but you can point them to another variable.
 ## Question 4
 
 ```js
-let weather;
-weather = "sunny";
+let weather;          // weather is undefined
+weather = "sunny";    //sunny is being assigned to the variable weather
```
Comment (berziiii) [inline diagnostic.md:59]: Correct.

File: diagnostic.md [hunk 5]
```diff
@@ -56,7 +74,10 @@ let answer = 'not empty';
 
 while (answer !== '') {
   answer = ask("Guess my secret? ");
-  if (answer === 'SeCrEt') {
+  if (answer < 1 || answer > 10) {
+    console.log("guess needs to be between 1 and 10 inclusive");
+  }
+  if (answer === '3') {
```
Comment (berziiii) [inline diagnostic.md:80]: Correct.

**Comment by berziiii** [comment]
5 / 5. Great work!

**Pull request closed without merging**